package/src/custom/src/application_campaign/ApplicationCampaignManager.php

```diff
@@ -0,0 +1,61 @@
+<?php
+// Copyright 2018 SugarCRM Inc.  Licensed by SugarCRM under the Apache 2.0 license.
+namespace Sugarcrm\Sugarcrm\custom\application_campaign;
+
+/**
+ * Class ApplicationCampaignManager
+ *
+ * Manages the Online Applications Campaign
+ *
+ * @package Sugarcrm\Sugarcrm\custom\application_campaign
+ */
+class ApplicationCampaignManager
+{
+
+
+    /**
+     * Get the most recently updated Campaign with the name 'Online Applications'
+     *
+     * Used with `getOnlineApplicationsCampaign` public custom entry point
+     *
+     * @return the id of the Online Applications Campaign
+     */
+    public function getOnlineApplicationsCampaignId(): string
+    {
+        // Get the most recently updated Campaign with the name 'Online Applications'
+        $q = $this->getSugarQuery();
+        $q->select(array('id', 'name'));
+        $q->from($this->getNewCampaignBean(), array('team_security' => false));
+        $q->where()->equals('name', 'Online Applications');
+        $q->orderBy('date_modified', 'desc');
+        $q->limit(1);
+
+        $results = $q->execute();
+
+        if(sizeof($results) != 1 || !$results[0]['id']){
+            throw new \Exception("Unable to find ID for the Campaign named Online Applications");
```

Comment:
Does this message get displayed to user? If so, we should consider using a language label.

Reply:
Good question!  I tried to find a way to display this message to the user, but I couldn't figure out how.  No matter what I did, the page just displayed "We are sorry, the server is currently unavailable, please try again later." and the error message never displayed in the console.  Is there a way to push a more informative message to the browser?

Reply:
Yeah, you could `echo` a message/HTML here which should get displayed in browser. Throwing an exception will cause server to treat it like an HTTP error response. This exception message is likely only placed in php error log.

Reply:
I played around with this some more.  If I echo a message, it gets sent back as the Campaign ID, which isn't great.  I decided to add some extra error handling to the Application Form to display an alert if the Campaign ID comes back empty (this would happen if someone installed Prof M without inserting the sample data).

Reply:
If you convert this to a REST API then you can throw a SugarApiException or create your own. That'll return a JSON encoded error message that'll allow a consumer to do error handling.